### PR TITLE
Add assay contacts

### PR DESCRIPTION
**Pull request by omaus**
- `arc assay person` functions added:
  - `register`: Register a person to an assay
  - `unregister`: Delete a person from an assay
  - `update`: Update a person from an assay with new metadata given
  - `edit`: Open a person from an assay and edit their metadata
  - `show`: Show the metadata of a person from an assay
  - `list`: Show all persons from all assays in this ARC
- some minor improvements on phrasing and readability included

### CODE REVIEW

File: src/ArcCommander/APIs/AssayAPI.fs [hunk 1]
```diff
@@ -691,327 +692,231 @@ module AssayAPI =
...
-            //    investigation
-            //|> Investigation.toFile investigationFilePath
-        
-        /// Opens an existing person by fullname (lastName, firstName, MidInitials) in the ARC investigation study with the text editor set in globalArgs.
-        let edit (arcConfiguration : ArcConfiguration) (personArgs : Map<string,Argument>) =
+            let persons = MetaData.getPersons "Investigation" doc
+
+            if API.Person.existsByFullName firstName midInitials lastName persons then
+                API.Person.updateByFullName updateOption person persons
+                |> MetaData.overwriteWithPersons "Investigation" <| doc
```
Comment (HLWeil) [inline src/ArcCommander/APIs/AssayAPI.fs:738]: I think backward piping (BWP) is rarely used and therefore might decrease readability for most users. Maybe you could bind the above value first

File: src/ArcCommander/APIs/AssayAPI.fs [hunk 2]
```diff
@@ -691,327 +692,231 @@ module AssayAPI =
...
+            else
+                if verbosity >= 1 then printfn "Person with the name %s %s %s does not exist in the assay with the identifier %s." firstName midInitials lastName assayIdentifier
+                if containsFlag "AddIfMissing" personArgs then
+                    if verbosity >= 1 then printfn "Registering person as AddIfMissing Flag was set." 
+                    API.Person.add persons person
+                    |> MetaData.overwriteWithPersons "Investigation" <| doc
```
Comment (HLWeil) [inline src/ArcCommander/APIs/AssayAPI.fs:744]: BWP (see above)

File: src/ArcCommander/APIs/AssayAPI.fs [hunk 3]
```diff
@@ -691,327 +692,231 @@ module AssayAPI =
...
+            let persons = MetaData.getPersons "Investigation" doc
+
+            match API.Person.tryGetByFullName firstName midInitials lastName persons with
+            | Some person ->
+                ArgumentProcessing.Prompt.createIsaItemQuery editor workDir
+                    (List.singleton >> Contacts.toRows None) 
+                    (Contacts.fromRows None 1 >> fun (_,_,_,items) -> items.Head)
+                    person
+                |> fun p -> API.Person.updateBy ((=) person) API.Update.UpdateAll p persons
+                |> MetaData.overwriteWithPersons "Investigation" <| doc
```
Comment (HLWeil) [inline src/ArcCommander/APIs/AssayAPI.fs:778]: BWP (see above)

File: src/ArcCommander/APIs/AssayAPI.fs [hunk 4]
```diff
@@ -691,327 +692,231 @@ module AssayAPI =
...
-            //    investigation
-            //|> Investigation.toFile investigationFilePath
-    
+            let assayFilePath = IsaModelConfiguration.tryGetAssayFilePath assayIdentifier arcConfiguration |> Option.get
+            
+            let doc = Spreadsheet.fromFile assayFilePath true
+            
+            let persons = MetaData.getPersons "Investigation" doc
+
+            API.Person.add persons person
```
Comment (HLWeil) [inline src/ArcCommander/APIs/AssayAPI.fs:824]: Check if person already exists. E.g. `arc s person register` fails if person with the same name already exists. This should be handled consistently.

File: src/ArcCommander/APIs/AssayAPI.fs [hunk 5]
```diff
@@ -691,327 +692,231 @@ module AssayAPI =
...
+            |> MetaData.overwriteWithPersons "Investigation" <| doc
```
Comment (HLWeil) [inline src/ArcCommander/APIs/AssayAPI.fs:825]: BWP (see above)

File: src/ArcCommander/APIs/AssayAPI.fs [hunk 6]
```diff
@@ -691,327 +692,231 @@ module AssayAPI =
...
+            let assayFilePath = IsaModelConfiguration.tryGetAssayFilePath assayIdentifier arcConfiguration |> Option.get
+
+            let doc = Spreadsheet.fromFile assayFilePath true
 
-            //// TO DO: Implementation für Assay-Datenstruktur
+            let persons = MetaData.getPersons "Investigation" doc
+
+            if API.Person.existsByFullName firstName midInitials lastName persons then
+                API.Person.removeByFullName firstName midInitials lastName persons
+                |> MetaData.overwriteWithPersons "Investigation" <| doc
```
Comment (HLWeil) [inline src/ArcCommander/APIs/AssayAPI.fs:851]: BWP (see above)

**Review by HLWeil — CHANGES_REQUESTED** [review]
Could you also check what happens if a command fails between opening and closing the assay xlsx file?
If the file stays blocked by the stream, a `try .. finally` might be necessary

**Comment by omaus** [comment]
> Could you also check what happens if a command fails between opening and closing the assay xlsx file? If the file stays blocked by the stream, a `try .. finally` might be necessary

Atm. it seems that it is not possible for the code to fail since all cases are covered and the AC prints a message with closing all file streams instead of failing. But this does _not_ mean that it is impossible in general.
Therefore, I will add `try finally`s.

**Review by HLWeil — APPROVED** [review]

**Pull request merged**